### PR TITLE
Test previously failing combos

### LINKED ISSUE #28: Have Erlang/OTP 17-19 work with Ubuntu 18.04
For history look at this one: https://github.com/hexpm/bob/issues/82.

### PR DESCRIPTION
**Pull request by paulo-ferraz-oliveira**

### CODE REVIEW

**Comment by paulo-ferraz-oliveira** [comment]
Yay, all passing. 👍

**Review by starbelly — APPROVED** [review]

**Comment by paulo-ferraz-oliveira** [comment]
I'm updating the compat. matrix.

**Comment by ericmj** [comment]
How did you test if ssl works?

**Comment by paulo-ferraz-oliveira** [comment]
> How did you test if ssl works?

I didn't, specifically. That said, these combos didn't pass even `mix -v`. Nice catch, though, lemme had that to the `rebar3` tests.

File: test-projects/mix/test/test_test.exs [hunk 1]
```diff
@@ -6,4 +6,8 @@ defmodule TestTest do
     out = Logfmt.encode(foo: "bar")
     assert out == "foo=bar"
   end
+
+  test "ensures ssl" do
+    assert {:ok, _} = Application.ensure_all_started(:ssl)
```
Comment (paulo-ferraz-oliveira) [inline test-projects/mix/test/test_test.exs:11]: @ericmj, would this be enough? I did the same on the `rebar3` test, but it's mostly redundant.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -57,8 +57,6 @@ jobs:
             otp-version: '22'
           - elixir-version: 'master'
             otp-version: '^22'
-          - elixir-version: 'master'
```
Comment (paulo-ferraz-oliveira) [inline .github/workflows/test.yml:60]: This was repeated two combos above. Also, I'm not sure all these combos are actually required (though they did help me during dev.)

File: test-projects/mix/test/test_test.exs (hunk 1 above)
Comment (ericmj) [reply test-projects/mix/test/test_test.exs:11]: Yes, that should be enough.

**Comment by paulo-ferraz-oliveira** [comment]
I trimmed down the number of tests, concentrated on non-versioning specific and added a few Ubuntus as per the compatibility matrix.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -21,99 +21,114 @@ jobs:
...
         run: |
           cd test-projects/mix
           mix deps.get
           mix test
-        if: ${{matrix.pair.elixir-version}}
+        if: ${{matrix.combo.elixir-version}}
       - name: Run rebar3 project tests
         run: |
           cd test-projects/rebar3
-          rebar3 as test do xref,dialyzer,ct,cover
```
Comment (paulo-ferraz-oliveira) [inline .github/workflows/test.yml:118]: Make tests _go_ faster.

**Comment by paulo-ferraz-oliveira** [comment]
@ericmj, is there still something you want to check here? Shall we go ahead and merge?

**Review by ericmj — APPROVED** [review]
I still think we are testing too many versions but other than that it looks good.

**Comment by paulo-ferraz-oliveira** [comment]
> I still think we are testing too many versions but other than that it looks good.

I'm OK with reducing. Tell me what you'd prefer, either as a general rule or by pointing out the ones you'd want removed, 1 by 1. I'm OK with either direction. 👍

**Comment by paulo-ferraz-oliveira** [comment]
Moving to merging this. We can revisit the high number of builds later if deemed required (if they're parallel, time will not be a big issue, I believe).

**Pull request merged**